swig/cpp/src/Tree_Schema.cpp

```diff
@@ -281,6 +281,7 @@ S_Set Schema_Node::xpath_atomize(enum lyxp_node_type ctx_node_type, const char *
         return nullptr;
     }
 
+    S_Deleter new_deleter = std::make_shared<Deleter>(set, deleter);
     return std::make_shared<Set>(set, deleter);
```

Comment:
this should likely return a shared_ptr with `new_deleter` instead of the old deleter, right?

Reply:
Yes, you are right, I'll fix it.